web_poet/requests.py

```diff
@@ -0,0 +1,120 @@
+"""
+Quick Start
+-----------
+
+.. code-block:: python
+
+    from web_poet import requests
+
+Developers could then use this alongside the ``async/await`` syntax in Python
+to issue additional requests.
+
+
+Use Cases
+---------
+
+Additional requests in Page Objects are hard to avoid since webpages nowadays
+requires some dynamic interactions. Some examples would be:
+
+    - "clicking" a load more button that loads more data.
+    - "scrolling" to paginate an infinite page.
+    - "hovering" that reveals a tool-tip containing additional page info.
+    - etc
+
+In most cases, these are done via AJAX requests.
+
+.. warning::
+
+    Additional requests should not be confused with "crawling" which aims to
+    visit multiple webpages from a given website. Additional requests are simply
+    a means to interact with the website to access more information from it.
+
+
+Use for other frameworks
+------------------------
+
+Please note that on its own, ``web_poet.request`` doesn't do anything. It doesn't
+know how to implement the request on its own. Thus, for frameworks or projects
+wanting to use additional requests in Page Objects, they need to set the
+implementation of how to download things via:
+
+.. code-block:: python
+
+    web_poet.request_backend_var.set(downloader_implementation)
+"""
+
+import asyncio
+import logging
+from contextvars import ContextVar
+from typing import Optional, List, Dict, ByteString, Any, Union
+
+import attr
+
+from web_poet.page_inputs import ResponseData
+
+logger = logging.getLogger(__name__)
+
+
+mapping = Dict[Union[str, ByteString], Any]
```

Comment:
Why is header value Any? Shouldn't it be str or bytes?

Reply:
Good catch! It was supposed to be `AnyStr`.

Let's discuss this further in the separate PR in https://github.com/scrapinghub/web-poet/pull/24.